plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -1356,13 +1365,35 @@ private function run_query(): void {
 		}
 
 		if ( $this->limits ) {
-			$this->found_orders  = absint( $wpdb->get_var( $this->count_sql ) ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+			$this->found_orders  = $this->get_count();
 			$this->max_num_pages = (int) ceil( $this->found_orders / $this->args['limit'] );
 		} else {
 			$this->found_orders = count( $this->orders );
 		}
 	}
 
+	/**
+	 * Get the order count by cache or SQL.
+	 */
+	private function get_count() {
+		global $wpdb;
+
+		$where_string  = substr( $this->count_sql, strpos( $this->count_sql, 'WHERE 1=1 AND' ) + 14, -1 );
+		$where_clauses = explode( ' AND ', $where_string );
+
+		// We can only use the aggregate cache when the filters are limited to type and status.
+		if (
+			count( $where_clauses ) === 2 &&
+			str_starts_with( $where_clauses[0], "({$wpdb->prefix}wc_orders.status" ) &&
+			str_starts_with( $where_clauses[1], "({$wpdb->prefix}wc_orders.type" )
```

Comment:
Suggession: `str_starts_with( $where_clauses[1], "({$wpdb->prefix}wc_orders.type = " )`

Reply:
All logic has been pulled out of `OrdersTableQuery` so we don't need to mess with clauses anymore.  There's still some small risk of a query being filtered manually, but this is likely a more extreme edge case.